### PR TITLE
Hierarchical model with correlated varying coefficient

### PR DESCRIPTION
**Pull request by lionel68**
Based on this forum post: https://forum.greta-stats.org/t/how-to-use-lkj-prior/175/2, a new example to add to the website for a hierarchical model with correlated varying coefficient.

### CODE REVIEW

**Comment by lionel68** [comment]
The commenting of the example code might be improved I guess ... Especially line 16:21.

File: inst/examples/hierarchical_linear_slopes_corr.Rmd [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+modmat <- model.matrix(~ Sepal.Width, iris) 
+# index of species
+jj <- as.numeric(iris$Species)
+
+M <- ncol(modmat) # number of varying coefficients
+N <- max(jj) # number of species
+
+# define the varying coefficient
+tau <- exponential(0.5, dim = M)
+Omega <- lkj_correlation(3, M)
```
Comment (lionel68) [inline inst/examples/hierarchical_linear_slopes_corr.Rmd]: 3 could be replaced by N here

**Comment by goldingn** [comment]
Ugh, thanks @lintr-bot. I shall deal with you tomorrow.

File: inst/examples/hierarchical_linear_slopes_corr.Rmd (hunk 1 above)
Comment (goldingn) [reply inst/examples/hierarchical_linear_slopes_corr.Rmd]: 3 is the shrinkage parameter, right? Any reason that should be set to N?

**Comment by goldingn** [comment]
@lionel68 could you please also the following chunk into  `vignettes/example_models.Rmd` in the relevant section - probably at the end of the 'Common models' section, following the uncorrelated one?

````
<hr>

```{r child='../inst/examples/hierarchical_linear_slopes_corr.Rmd'}
```
````

File: inst/examples/hierarchical_linear_slopes_corr.Rmd (hunk 1 above)
Comment (goldingn) [inline inst/examples/hierarchical_linear_slopes_corr.Rmd]: Could you please add a comment about here saying that the following code is a more computationally efficient and stable way of sampling `ab` as a multivariate normal, with covariance Sigma

Comment (lionel68) [reply inst/examples/hierarchical_linear_slopes_corr.Rmd]: Right, my bad ...

**Comment by lionel68** [comment]
Right, I changed a bit the commenting based on what I understand of this parametrization, let me know if there are some mistakes / misunderstanding.

**Comment by goldingn** [comment]
Looks great, thanks! I'll merge and rebuild the website tomorrow. I'll add you to the Description file as a contributor too.

**Pull request merged**